src/Components/PelcroModalController/PelcroModalController.service.js

```diff
@@ -272,6 +281,53 @@ export const initSubscriptionFromURL = () => {
   });
 };
 
+/**
+ * Initializes offline subscription flow if 'plan_id' params exist
+ * with valid IDs, Otherwise, ignore the param
+ */
+export const initOfflineSubscriptionFromURL = (offlinePlanId) => {
+  const {
+    switchView,
+    whenSiteReady,
+    isAuthenticated,
+    switchToPaymentView,
+    switchToAddressView,
+    set
+  } = usePelcro.getStore();
+
+  whenSiteReady(() => {
+    window.Pelcro.plan.getPlan(
+      {
+        plan_id: offlinePlanId
+      },
+      (error, response) => {
+        if (error) {
+          return;
+        }
+
+        const { plan } = response.data;
+
+        set({
+          plan,
+          product: plan?.product
+        });
+
+        if (!isAuthenticated()) {
+          return switchView("register");
+        }
+
+        const requiresAddress = Boolean(plan.address_required);
```

Comment:
This should be plan.product.address_required, right? Not sure if they're the same